tracing/tracing.go

```diff
@@ -0,0 +1,107 @@
+package tracing
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// propagator to use
+var propagator propagation.TextMapPropagator = propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}) // traceparent header
+type Option interface {
+	apply(*options)
+}
+type options struct {
+	address string
+}
+
+// WithAddress takes an address in the form of Host:Port
+func WithAddress(addr string) Option {
+	return addrOption{address: addr}
+}
+
+type addrOption struct {
+	address string
+}
+
+func (o addrOption) apply(opts *options) {
+	opts.address = o.address
+}
+
+// OtlpGrpcExporter uses the otlptracegrpc modules and the otlptrace module to produce a new exporter at our default addr
+func OtlpGrpcExporter(ctx context.Context, opts ...Option) (sdktrace.SpanExporter, error) {
+
+	const DefaultCollectorHost = "localhost"
```

Comment:
ah sorry, meant these should be defined at the top of the file, with an explanation

Reply:
I'm curious, why? I would think it would make sense to keep them in the context of the function, but I have seen that convention of defining consts at the top before.

Reply:
If a developer is ever trying to find default values, it is much easier to first check at the top of the file instead of digging through functions